### PR TITLE
Azure AD B2C Permissions

### PR DESCRIPTION
**Pull request by zahalzel**
Updates for configuring Azure AD B2C:
- Adds Microsoft Graph scopes for all B2C apps
- Adds --api-scopes CLI option to input scopes for B2C Web API scenarios
- Adds Permissions and Scopes and Admin Consent for B2C Web API scenarios

### CODE REVIEW

File: src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/AuthenticationParameters/AzureAdProperties.cs [hunk 1]
```diff
@@ -110,9 +112,7 @@ public AzureAdBlock(ApplicationParameters applicationParameters, JObject? existi
             TenantId = TenantId ?? DefaultProperties.TenantId,
             ClientId = ClientId ?? DefaultProperties.ClientId,
             CallbackPath = CallbackPath ?? DefaultProperties.CallbackPath,
-            Scopes = Scopes ?? DefaultProperties.ApiScopes,
-            ClientSecret = ClientSecret ?? DefaultProperties.ClientSecret,
```
Comment (deepchoudhery) [inline src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/AuthenticationParameters/AzureAdProperties.cs:114]: why do we not need these properties anymore? or if they are initialized elsewhere, why?

File: src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/MicrosoftIdentityPlatform/MicrosoftIdentityPlatformApplicationManager.cs [hunk 2]
```diff
@@ -135,6 +126,21 @@ public class MicrosoftIdentityPlatformApplicationManager
             return effectiveApplicationParameters;
         }
 
+        private static string GetMsGraphScopes(ApplicationParameters applicationParameters)
+        {
+            var apiScopes = applicationParameters.CalledApiScopes ?? string.Empty;
+            if (!apiScopes.Contains("openid"))
```
Comment (deepchoudhery) [inline src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/MicrosoftIdentityPlatform/MicrosoftIdentityPlatformApplicationManager.cs:132]: does case matter here?

File: src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/MicrosoftIdentityPlatform/MicrosoftIdentityPlatformApplicationManager.cs [hunk 3]
```diff
@@ -386,56 +444,55 @@ internal static bool UpdateImplicitGrantSettings(Application app, ProvisioningTo
...
-                    ResourceId = createdServicePrincipal.Id,
-                    Scope = DefaultProperties.ApiScopes
-                };
-
-                await graphServiceClient.Oauth2PermissionGrants
-                    .Request()
-                    .AddAsync(oAuth2PermissionGrant).ConfigureAwait(false);
-            }
-        }
+        //private async Task AddApiPermissionFromBlazorwasmHostedSpaToServerApi(
```
Comment (deepchoudhery) [inline src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/MicrosoftIdentityPlatform/MicrosoftIdentityPlatformApplicationManager.cs:447]: Let's remove the commented out method. If we need part of this later, we can come back to this PR.

**Review by deepchoudhery — CHANGES_REQUESTED** [review]
Looks good, couple small inquiries and changes.

File: src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/AuthenticationParameters/AzureAdProperties.cs (hunk 1 above)
Comment (zahalzel) [reply src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/AuthenticationParameters/AzureAdProperties.cs:114]: Yeah they are only necessary when the app registration has a client secret (when a downstream API or MS graph is called), so I moved these properties to a separate object (DownstreamAiSettings)

**Pull request merged**